src/com/goide/GoDocumentationProvider.java

```diff
@@ -44,7 +44,14 @@ public String generateDoc(PsiElement element, PsiElement originalElement) {
       Collection<PsiElement> children = PsiTreeUtil.findChildrenOfType(topLevel, element.getClass());
       boolean alone = children.size() == 1 && children.iterator().next().equals(element);
       List<PsiComment> comments = getPreviousNonWsComment(alone ? topLevel : element);
-      if (!comments.isEmpty()) return getCommentText(comments);
+      String result = "<pre>" + getSignature(element);
+      result = result.
+        replaceAll("\n", "").
+        replaceAll("\\s+", " ").
+        replaceAll("\\(\\s", "(").
+        replaceAll(",\\s\\)", ")");
```

Comment:
It's not a nice way to create a one-liners. Please use PSI for this. See GoParameterInfoHandler.